src/main/java/gregtech/api/mui/sync/appeng/AEFluidSyncHandler.java

```diff
@@ -0,0 +1,81 @@
+package gregtech.api.mui.sync.appeng;
+
+import gregtech.api.mui.GTByteBufAdapters;
+import gregtech.api.util.GTUtility;
+import gregtech.api.util.JEIUtil;
+import gregtech.common.metatileentities.multi.multiblockpart.appeng.slot.ExportOnlyAEFluidList;
+import gregtech.common.metatileentities.multi.multiblockpart.appeng.slot.ExportOnlyAEFluidSlot;
+import gregtech.common.metatileentities.multi.multiblockpart.appeng.slot.IConfigurableSlot;
+import gregtech.common.metatileentities.multi.multiblockpart.appeng.stack.WrappedFluidStack;
+
+import net.minecraftforge.fluids.FluidStack;
+import net.minecraftforge.fml.relauncher.Side;
+import net.minecraftforge.fml.relauncher.SideOnly;
+
+import appeng.api.storage.data.IAEFluidStack;
+import com.cleanroommc.modularui.utils.serialization.IByteBufAdapter;
+import it.unimi.dsi.fastutil.ints.Int2ObjectMap;
+import mezz.jei.api.gui.IRecipeLayout;
+import mezz.jei.api.recipe.transfer.IRecipeTransferError;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class AEFluidSyncHandler extends AESyncHandler<IAEFluidStack> {
+
+    protected final ExportOnlyAEFluidList fluidList;
+
+    public AEFluidSyncHandler(ExportOnlyAEFluidList fluidList, @Nullable Runnable dirtyNotifier) {
+        super(fluidList.getInventory(), fluidList.isStocking(), dirtyNotifier);
+        this.fluidList = fluidList;
+    }
+
+    @Override
+    protected @NotNull IConfigurableSlot<IAEFluidStack> @NotNull [] initializeCache() {
+        // noinspection unchecked
+        IConfigurableSlot<IAEFluidStack>[] cache = new IConfigurableSlot[slots.length];
+        for (int index = 0; index < slots.length; index++) {
+            cache[index] = new ExportOnlyAEFluidSlot();
+        }
+        return cache;
+    }
+
+    @Override
+    protected @NotNull IByteBufAdapter<IAEFluidStack> initializeByteBufAdapter() {
+        return GTByteBufAdapters.WRAPPED_FLUID_STACK;
+    }
+
+    @Override
+    public boolean isStackValidForSlot(int index, @Nullable IAEFluidStack stack) {
+        if (stack == null) return true;
+        if (!isStocking) return true;
+        return !fluidList.hasStackInConfig(((WrappedFluidStack) stack).getDelegate(), true);
+    }
+
+    @Override
+    public IRecipeTransferError receiveRecipe(@NotNull IRecipeLayout recipeLayout, boolean maxTransfer,
```

Comment:
Can this be made to work with fluid containers in a recipe input? EG, our treated wood recipe with a bucket of creosote. Could the recipe transfer take the contained fluid in a fluid handler and then use that?